plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/conversions/RemoveWrongExtraModifiersConversion.kt

```diff
@@ -6,14 +6,23 @@ import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
 import org.jetbrains.kotlin.nj2k.RecursiveConversion
 import org.jetbrains.kotlin.nj2k.tree.*
 
-internal class RemoveWrongExtraModifiersForSingleFunctionsConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
+internal class RemoveWrongExtraModifiersConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
     override fun applyToElement(element: JKTreeElement): JKTreeElement {
         if (element !is JKOtherModifiersOwner) return recurse(element)
         if (element.parentOfType<JKClass>() == null) {
+            // single function --> remove STATIC
             element.elementByModifier(OtherModifier.STATIC)?.also { modifierElement ->
                 element.otherModifierElements -= modifierElement
             }
         }
+        else if ( element is JKClass && element.classKind == JKClass.ClassKind.OBJECT && element.parents()
+                .none { parent -> parent is JKClass && parent.classKind == JKClass.ClassKind.OBJECT }) {
+            // Object not nested in another object --> remove INNER
+            element.elementByModifier(OtherModifier.INNER)?.also { modifierElement ->
```

Comment:
You need to determine which modifier to remove. The removal part itself is duplicated code, can be refactored.